compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -262,7 +262,7 @@ pub fn each_linked_rlib(
     };
 
     for &cnum in crates {
```

Comment:
Suggestion: can you move the `let crates = info.used_crates.iter();` next to this for loop? AFAICT `crates` is not used in the `fmts` above. Maybe even pull out the `fmts` logic into a helper (maybe `dependency_formats()`?) if that makes the abstraction levels less mixed.